File: src/me/krickl/memebotj/InternalCommands/UptimeCommand.java
package me.krickl.memebotj.InternalCommands;

import me.krickl.memebotj.ChannelHandler;
import me.krickl.memebotj.CommandHandler;
import me.krickl.memebotj.UserHandler;

public class UptimeCommand extends CommandHandler {
	
	public UptimeCommand(String channel, String command, String dbprefix) {
		super(channel, command, dbprefix);
		this.setNeededCommandPower(75);
	}

	@Override
	public void commandScript(UserHandler sender, ChannelHandler channelHandler, String[] data) {
		int currentUpTimeSeconds = channelHandler.getStreamStartTime() - (int) (System.currentTimeMillis() / 1000L);
		channelHandler.sendMessage(String.format("Uptime: %d", currentUpTimeSeconds), this.getChannelOrigin());
	}
}

File: src/me/krickl/memebotj/InternalCommands/UptimeCommand.scala
package me.krickl.memebotj.InternalCommands

import me.krickl.memebotj.ChannelHandler
import me.krickl.memebotj.CommandHandler
import me.krickl.memebotj.UserHandler

class UptimeCommand(channel: String, command: String, dbprefix: String) extends CommandHandler(channel: String, command: String, dbprefix: String) {
  this.setNeededCommandPower(75)

  override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
    val currentUpTimeSeconds: Int = channelHandler.getStreamStartTime - (System.currentTimeMillis / 1000L).toInt
    channelHandler.sendMessage(f"Uptime: ${currentUpTimeSeconds}", this.getChannelOrigin)
  }
}